shell/platform/android/io/flutter/plugin/editing/TextEditingDelta.java

```diff
@@ -0,0 +1,200 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugin.editing;
+
+import androidx.annotation.VisibleForTesting;
+import io.flutter.Log;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/// A representation of the change that occured to an editing state, along with the resulting
+/// composing and selection regions.
+public class TextEditingDelta {
+  private CharSequence oldText;
+  private CharSequence deltaText;
+  private CharSequence deltaType;
+  private int deltaStart;
+  private int deltaEnd;
+  private int newSelectionStart;
+  private int newSelectionEnd;
+  private int newComposingStart;
+  private int newComposingEnd;
+
+  private static final String TAG = "TextEditingDelta";
+
+  public TextEditingDelta(
+      CharSequence oldEditable,
+      CharSequence newEditable,
+      int start,
+      int end,
+      CharSequence tb,
+      int tbstart,
+      int tbend,
+      int selectionStart,
+      int selectionEnd,
+      int composingStart,
+      int composingEnd) {
+    newSelectionStart = selectionStart;
+    newSelectionEnd = selectionEnd;
+    newComposingStart = composingStart;
+    newComposingEnd = composingEnd;
+    final boolean isDeletionGreaterThanOne = end - (start + tbend) > 1;
+    final boolean isCalledFromDelete = tb == "" && tbstart == 0 && tbstart == tbend;
```

Comment:
is the `tb == ""` check because of the `delete` implementation here?
https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/text/SpannableStringBuilder.java;l=230-237?q=spannablestringbuilder

Reply:
Yeah that's where I got that from.